shell/shared/db.js

```diff
@@ -86,6 +86,46 @@ Grains = new Mongo.Collection("grains");
 //   publicId:  An id used to publicly identify this grain. Used e.g. to route incoming e-mail and
 //       web publishing. This field is initialized when first requested by the app.
 
+RoleAssignments = new Mongo.Collection("roleAssignments");
+// Edges in the permissions sharing graph.
+//
+// To share a permission with another user is to declare "if I have this permission, then so should
+// the recipient". A bundle of such declarations by a single sharer directed at a single recipient
+// for a single grain is called a "role assignment". A grain's owner always has every permission,
+// but permissions for other users must be computed from this collection.
+//
+// Every role assignment implicitly shares the special "can view grain" permission, which dictates
```

Comment:
This comment could be confusing in that "can view grain" could be taken to mean "has read-only access to grain content", but here I think you only mean "can tell that the grain exists".

I'm not really sure I'd describe this as a permission. I might say something like: "Receiving any role assignment causes the grain to appear in the user's file list, even if the assignment provides no actual permissions."